server/graphql/v2/mutation/AccountMutations.ts

```diff
@@ -159,9 +174,17 @@ const accountMutations = {
 
       const encryptedText = crypto.encrypt(args.token);
 
-      await user.update({ twoFactorAuthToken: encryptedText });
+      /** Generate recovery codes, hash and store them in the table, and return them to the user to write down */
+      const recoveryCodesArray = Array.from({ length: 6 }, () =>
+        cryptoRandomString({ length: 16, type: 'distinguishable' }),
+      );
+      const hashedRecoveryCodesArray = recoveryCodesArray.map(code => {
+        return crypto.hash(code);
+      });
```

Comment:
I don't think we need to hash them, we might want to encrypt the DB field tho.
Also, there's no purpose in using distinguishable codes if we're going to hash it later on.

Reply:
We were going to just encrypt them but @znarf has requested for them to be hashed for extra security. https://github.com/opencollective/opencollective/issues/3709#issuecomment-746120617

Reply:
I haven't asked for **extra** hashing, I just asked for hashing. Why?

Hashing is more secure than encrypting, as it's one way and can't be decoded.

Whenever you can use hashing instead of encrypting always use hashing. We're currently using encryption for cases when hashing is not practical, when data needs to be decrypted.

There is no need to add encryption on top of hashing, it's a useless layer showing you're not 100% comfortable with hashing as a security measure.

Reply:
The code only hashes them now anyways. Was momentarily confused by the old discussion on the issue.